src/plugins/intel_gpu/tests/unit/transformations/lora_horizontal_fusion.cpp

```diff
@@ -0,0 +1,171 @@
+        auto split_axis = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{}, {2});
+        auto split = std::make_shared<ov::op::v1::Split>(multiply, split_axis, 3);
+
+        auto variable_b_0 = std::make_shared<ov::op::util::Variable>(
+            ov::op::util::VariableInfo{ov::PartialShape({2048, -1}), model_dt, "var_b_0"});
+        auto variable_b_1 = std::make_shared<ov::op::util::Variable>(
+            ov::op::util::VariableInfo{ov::PartialShape({256, -1}), model_dt, "var_b_1"});
+        auto variable_b_2 = std::make_shared<ov::op::util::Variable>(
+            ov::op::util::VariableInfo{ov::PartialShape({256, -1}), model_dt, "var_b_2"});
+
+        auto read_value_b_0 = std::make_shared<ov::op::v6::ReadValue>(variable_b_0);
+        auto read_value_b_1 = std::make_shared<ov::op::v6::ReadValue>(variable_b_1);
+        auto read_value_b_2 = std::make_shared<ov::op::v6::ReadValue>(variable_b_2);
+
+        auto matmul2_0 = std::make_shared<ov::op::v0::MatMul>(split->output(0), read_value_b_0, false, true);
+        auto matmul2_1 = std::make_shared<ov::op::v0::MatMul>(split->output(1), read_value_b_1, false, true);
+        auto matmul2_2 = std::make_shared<ov::op::v0::MatMul>(split->output(2), read_value_b_2, false, true);
+
+        auto concat_matmul2 = std::make_shared<ov::op::v0::Concat>(NodeVector{matmul2_0, matmul2_1, matmul2_2}, 2);
+
+        auto add = std::make_shared<ov::op::v1::Add>(fc_fused, concat_matmul2);
+
+        auto axis_const = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {2});
+        auto split_const = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{3}, {2048, 256, 256});
+        auto var_split = std::make_shared<ov::op::v1::VariadicSplit>(add, axis_const, split_const);
+
+        auto reshape_pattern0 = std::make_shared<ov::op::v0::Constant>(ov::element::i64, ov::Shape{4}, std::vector<int64_t>{0, 0, 32, 64});
+        auto reshape_pattern1 = std::make_shared<ov::op::v0::Constant>(ov::element::i64, ov::Shape{4}, std::vector<int64_t>{0, 0, 4, 64});
+        auto reshape_pattern2 = std::make_shared<ov::op::v0::Constant>(ov::element::i64, ov::Shape{4}, std::vector<int64_t>{0, 0, 4, 64});
+        auto reshape0 = std::make_shared<ov::op::v1::Reshape>(var_split->output(0), reshape_pattern0, true);
+        auto reshape1 = std::make_shared<ov::op::v1::Reshape>(var_split->output(1), reshape_pattern1, true);
+        auto reshape2 = std::make_shared<ov::op::v1::Reshape>(var_split->output(2), reshape_pattern2, true);
+
+        auto result0 = std::make_shared<ov::op::v0::Result>(reshape0);
+        auto result1 = std::make_shared<ov::op::v0::Result>(reshape1);
+        auto result2 = std::make_shared<ov::op::v0::Result>(reshape2);
+
+        model_ref = std::make_shared<ov::Model>(ov::NodeVector{result0, result1, result2}, ov::ParameterVector{lora_input});
+        comparator.enable(FunctionsComparator::ATTRIBUTES);
+        comparator.enable(FunctionsComparator::CmpValues::ACCURACY);
```

Comment:
I think we need a functional test for such kind of a graph as ACCURACY check doesn't work for dynamic models.

Reply:
Added func test